client/src/scenes/Plans.jsx

```diff
@@ -149,14 +358,45 @@ class Plans extends Component {
         }
     }
 
+    handleScrollEvent = (e) => {
+        if (this.lastScrollY > window.scrollY) {
+            this.setState({
+                hiddenTopSection: false,
+                translateY: -100,
+            });
+        } else if (this.lastScrollY < window.scrollY) {
+            this.setState({
+                hiddenTopSection: true,
+                hiddenTopContentSection: true,
+                translateY: -200,
+            });
+            console.log('down');
+        }
+        this.lastScrollY = window.scrollY;
+        if (window.scrollY === 0) {
+            this.setState({
+                hiddenTopSection: false,
+                hiddenTopContentSection: false,
+                translateY: -0,
+            });
+        }
+    };
+
     componentDidMount() {
+        this.lastScrollY = window.scrollY;
         // init location service
         locationAutocompleteApi.init();
 
         // if there is no valid query string params to search by run default search
         if (!this.loadQsSearchParams()) {
             this.loadPlans(this.state.pageNumber);
         }
+
+        document.addEventListener('scroll', this.handleScrollEvent);
```

Comment:
Maybe the're a react hook for that?

Reply:
@GalGend we cant use react hooks in the class components. componentDidMount it is like useEffect. if you want, I can rewrite PlansHeader as a functional component.